### PR TITLE
minor release process change

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

The minor release process is changing in order to not have major release changes sitting in the pull request tab.

This changes from taking master and releasing from master to creating a branch that you cherry-pick commits into. 

There are two options on labeling which pull requests to include in a minor release:
1. Use the label `R:minor` to know which pull requests to include then remove the label when those pull requests have been included in a release.
2. Create an Issue where pull request numbers are added. then the issue is closed when the release is done.

this process should be followed after 0.33.3

______

For contributor use:

- [ ] Wrote tests
- [ ] Updated CHANGELOG_PENDING.md
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -207,20 +210,50 @@ easy to reference the pull request where a change was introduced.
...
 
-If there were no breaking changes and you need to create a release nonetheless,
-the procedure is almost exactly like with a new release above.
+Minor releases are done differently from major releases. Minor release pull requests should be labeled with `R:minor` if they are to be included.
+
+1. Checkout the last major release, `vX.X`.
+
+   - `git checkout vX.X`
+
+2. Create a branch off the latest release, `rc1/vX.X.x`, and push the branch.
```
Comment (erikgrinaker) [inline CONTRIBUTING.md]: Why `rc1`?

**Comment by erikgrinaker** [comment]
Totally in support of having release branches. Note that Mergify has built-in support for backporting into release branches: https://doc.mergify.io/actions.html#backport

File: CONTRIBUTING.md (hunk 1 above)
Comment (marbar3778) [reply CONTRIBUTING.md]: why we need an rc branch or the name?

Comment (erikgrinaker) [reply CONTRIBUTING.md]: Nevermind. I took this to be that all minor releases should have a `rc1/` prefix.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -207,20 +210,50 @@ easy to reference the pull request where a change was introduced.
...
+1. Checkout the last major release, `vX.X`.
+
+   - `git checkout vX.X`
+
+2. Create a release candidate branch off the most recent major release with your upcoming version specified, `rc1/vX.X.x`, and push the branch.
+
+   - `git checkout -b rc1/vX.X.x`
+   - `git push -u origin rc1/vX.X.x`
+
+3. Create a cherry-picking branch, and make a pull request into the release candidate.
```
Comment (melekes) [inline CONTRIBUTING.md]: Can you explain again why do we need cherry-picking? I've only done it a few times while backporting stuff.

Comment (marbar3778) [reply CONTRIBUTING.md]: yes, in this process we are merging all pull requests to master, even breaking ones. In order to do minor releases, we will not cut from master as we have been doing in the past but rather cherry-pick the commits that are meant to go in a minor release.

This is in line with how the sdk is doing things.

Comment (melekes) [reply CONTRIBUTING.md]: Okay. Thank you

**Pull request merged**